release_notes/ocp-4-16-release-notes.adoc

```diff
@@ -311,6 +311,21 @@ With this release, the etcd tuning parameters can be set to values that optimize
 * `Standard`
 * `Slower`
 
+[discrete]
+[id="ocp-4-16-unauthenticated-users-cluster-role-bindings_{context}"]
+=== Unauthenticated users no longer have access to cluster role bindings
```

Comment:
### Role Bindings vs Roles

The `system:unauthenticated` was removed from cluster role bindings, which leads to the loss of access that was given by the cluster roles. So something like this would be more correct:

```
Unauthenticated users were removed from some access to cluster roles.
```

OR

```
Unauthenticated users were removed from some cluster role bindings.
```

### Unauthenticated still have access to certain roles

`system:unauthenticated` wasn't removed from access to all roles. We kept unauthenticated in two "public" roles. They contain public information that are also being exposed by default in kubernetes upstream.

So we could emphasize that we removed or narrowed down unnecessary access for unauthenticated groups or that we removed general access, which is only being used in certain use cases?

GitHub is a use-case, where you need to give up the k8s / ocp authentication as they don't support it. So users who would not want GitHub to trigger builds, had unnecessarily access for unauthenticated.

This isn't a vulnerability, as there is a custom solution by GitHub, that ensures authenticity. But the custom solution needs to by-pass k8s / ocp auth solutions. 
The solution from GitHub relies on a shared secret, that is being used to create a hash, that needs to be validated.

Reply:
#### Reasons

##### For removal

- Defense in Depth
- Isn't necessary for a default installation

##### Why it was there

- Lack of K8s support for Feature-Providers like GitHub.